services/viva-ms/lambdas/createVivaCase.js

```diff
@@ -1,191 +1,120 @@
 /* eslint-disable no-console */
 import to from 'await-to-js';
 import uuid from 'uuid';
-import { throwError } from '@helsingborg-stad/npm-api-error-handling';
 
 import config from '../../../config';
+
+import * as dynamoDB from '../../../libs/dynamoDb';
 import params from '../../../libs/params';
+import log from '../../../libs/logs';
 import { putItem } from '../../../libs/queries';
-import * as dynamoDB from '../../../libs/dynamoDb';
-import { CASE_PROVIDER_VIVA } from '../../../libs/constants';
-import { getStatusByType } from '../../../libs/caseStatuses';
-import validateApplicationStatus from '../helpers/validateApplicationStatus';
+import { getStatusByType, statusTypes } from '../../../libs/caseStatuses';
 import { populateFormWithPreviousCaseAnswers } from '../../../libs/formAnswers';
-
 import { getFutureTimestamp, millisecondsToSeconds } from '../../../libs/timestampHelper';
-import { DELETE_VIVA_CASE_AFTER_12_HOURS } from '../../../libs/constants';
-
-import vivaAdapter from '../helpers/vivaAdapterRequestClient';
-import log from '../../../libs/logs';
+import {
+  CASE_PROVIDER_VIVA,
+  DELETE_VIVA_CASE_AFTER_12_HOURS,
+  VIVA_CASE_CREATED,
+} from '../../../libs/constants';
 
 const VIVA_CASE_SSM_PARAMS = params.read(config.cases.providers.viva.envsKeyName);
 
 export async function main(event, context) {
-  const userDetail = event.detail;
+  const { clientUser, vivaPersonDetail } = event.detail;
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if (noSuchProperty('vivaPersonDetail', event.detail)){
  // do something
}

 if(noSuchProperty('clientUser', event.detail)) {
   // do something
 }

const {clientUser, vivaPersonDetail} = event.detail

```

Reply:
Will be fixed in future task